src/Docker/Service/Config.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Docker\Service;
+
+use Magento\MagentoCloud\Docker\Config\Reader;
+use Magento\MagentoCloud\Filesystem\FileSystemException;
+use Magento\MagentoCloud\Docker\ConfigurationMismatchException;
+use Illuminate\Contracts\Config\Repository;
+
+/**
+ * Retrieve Service versions/configs from Cloud configuration.
+ */
+class Config
+{
+    const KEY_PHP = 'php';
+    const KEY_DB = 'mysql';
+    const KEY_NGINX = 'nginx';
+    const KEY_REDIS = 'redis';
+    const KEY_ELASTICSEARCH = 'elasticsearch';
+    const KEY_RABBITMQ = 'rabbitmq';
+
+    const KEY_CRON = 'cron';
+    const KEY_VARNISH = 'varnish';
+
+    /**
+     * List of services which can be configured in Cloud docker
+     *
+     * @var array
+     */
+    private $configurableVersions = [
+        self::KEY_PHP,
+        self::KEY_DB,
+        self::KEY_NGINX,
+        self::KEY_REDIS,
+        self::KEY_ELASTICSEARCH,
+        self::KEY_RABBITMQ,
+    ];
+
+    /**
+     * @var Reader
+     */
+    private $reader;
+
+    /**
+     * @param Reader $reader
+     */
+    public function __construct(Reader $reader)
+    {
+        $this->reader = $reader;
+    }
+
+    /**
+     * Retrieves service versions set in configuration files.
+     * Returns null if neither of services is configured or provided in $customVersions.
+     *
+     * Example of return:
+     *
+     * ```php
+     *  [
+     *      'elasticserach' => '5.6',
+     *      'db' => '10.0'
+     *  ];
+     * ```
+     *
+     * @param Repository $customVersions overwrite version from config files
```

Comment:
`overwrite version from config files` looks like description isn't correct as ` $customVersions` contains services versions from `docker:build` command input.

Reply:
I meant that value from this argument will overwrite versions set in config files. 
is `custom version which overwrite values from configuration files` be more clear?

Reply:
yes, thank you